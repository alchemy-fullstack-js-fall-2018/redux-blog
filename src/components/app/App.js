import React, { Fragment } from 'react';
import Header from '../header/Header';
import {
  BrowserRouter as Router,
  Switch, Route
} from 'react-router-dom';
import { routerRoutes } from '../../routes';

export default function App() {
  console.log(routerRoutes);
  return (
    <Fragment>
      <h1>Welcome!</h1>
      <Header />
      <Router>
        <Switch>
          {routerRoutes()}
        </Switch>
      </Router>
    </Fragment>
  );
}
