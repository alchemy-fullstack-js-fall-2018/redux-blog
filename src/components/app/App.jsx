import React, { Fragment } from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import { ROUTES } from '../../routes';
import { Helmet } from 'react-helmet';
import { createGlobalStyle, ThemeProvider } from 'styled-components';
import Header from './Header.jsx';
import Footer from './Footer.jsx';

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: BOMBARD;
    src: url("/src/assets/fonts/BOMBARD.ttf") format("opentype");
  }
  #root {
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: space-between;
    text-align: center;
    margin: 0;
  }
  html {
    box-sizing: border-box;
    font-size: 10px;
  }
  *, *:before, *:after {
    box-sizing: inherit;
    font-family: sans-serif;
  }
  body {
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height: 2;
  }
`;

const theme = {
  primary: 'rgb(0, 153, 0)',
  secondary: 'rgb(204, 255, 255)',
  darksecondary: 'rgb(107, 48, 13)',
  accent: 'rgb(173, 38, 36)'
};

const App = () => {

  return (
    <ThemeProvider theme={theme}>
      <BrowserRouter>
        <Fragment>
          <Helmet>
            <title>Redux Blogging</title>
            <link rel="icon" href="/src/assets/favicon.ico"/>
          </Helmet>
          <GlobalStyle/>
          <Header />
          <Switch>
            <Route exact component={ROUTES.ABOUT.Component} path={ROUTES.ABOUT.path} />
            <Route exact component={ROUTES.HOME.Component} path={ROUTES.HOME.path} />
            <Route exact component={ROUTES.POSTS.Component} path={ROUTES.POSTS.path} />
            <Route exact component={ROUTES.POST_DETAIL.Component} path={ROUTES.POST_DETAIL.path} />
            <Route exact component={ROUTES.USER_REGISTER.Component} path={ROUTES.USER_REGISTER.path} />
            <Route exact component={ROUTES.USERS.Component} path={ROUTES.USERS.path} />
            <Route exact component={ROUTES.USER_DETAIL.Component} path={ROUTES.USER_DETAIL.path} />
            <Redirect to={ROUTES.HOME.path} />
          </Switch>
          <Footer />
        </Fragment>
      </BrowserRouter>
    </ThemeProvider>
  );
};

export default App;
