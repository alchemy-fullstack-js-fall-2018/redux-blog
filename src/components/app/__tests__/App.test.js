import React from 'react';
import { shallow } from 'enzyme';
import App from '../App.jsx';


describe('<App />', () => {
  test('renders', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
