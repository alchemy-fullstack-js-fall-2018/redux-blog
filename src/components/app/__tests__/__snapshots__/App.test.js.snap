// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<ThemeProvider
  theme={
    Object {
      "accent": "rgb(173, 38, 36)",
      "darksecondary": "rgb(107, 48, 13)",
      "primary": "rgb(0, 153, 0)",
      "secondary": "rgb(204, 255, 255)",
    }
  }
>
  <BrowserRouter>
    <HelmetWrapper
      defer={true}
      encodeSpecialCharacters={true}
    >
      <title>
        Redux Blogging
      </title>
      <link
        href="/src/assets/favicon.ico"
        rel="icon"
      />
    </HelmetWrapper>
    <GlobalStyleComponent />
    <Header />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/about/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/home/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/posts/search"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/posts/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/users/new"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/users"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/users/:id"
      />
      <Redirect
        push={false}
        to="/home/"
      />
    </Switch>
    <Footer />
  </BrowserRouter>
</ThemeProvider>
`;
